syntax="proto3";
//package grpc_hub;

message SignUpRequest {
  string name = 1;
  string email = 2;
  string username = 3;
  string password = 4;
  string confirm_password = 5;

}

message LogInRequest {
  string email = 1;
  string username = 2;
  string password = 3;
}

message User {
  int64 id = 1;
  string name = 2;
  string email = 3;
  string username = 4;
}

message LoginResponse {
  string token = 1;
  User user = 2;
}

service UserService{
  rpc SignUp(SignUpRequest) returns (LoginResponse) {}
  rpc Login(LogInRequest) returns (LoginResponse) {}
  rpc GetUser(User) returns (User) {}
}
